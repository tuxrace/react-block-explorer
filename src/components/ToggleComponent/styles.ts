const styles = {
    toggleStyle: {
        marginBottom: 16
    }
}

export default styles
