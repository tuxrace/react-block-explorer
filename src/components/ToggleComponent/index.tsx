import React, {useState} from 'react'
import ToggleButton from '@material-ui/lab/ToggleButton';
import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';
import { Grid, makeStyles } from '@material-ui/core';
import styles from './styles';
import { CoinTypes } from '../../types';
import { tokenMap } from '../../config';

const ToggleComponent = (props: any) => {
    const {handleChange, selected} = props
    const useStyles = makeStyles(styles)
    const classes = useStyles()
    
    return <ToggleButtonGroup exclusive value={selected} onChange={handleChange} className={classes.toggleStyle}>
        <ToggleButton value={tokenMap[CoinTypes.Alice]}> {CoinTypes.Alice} </ToggleButton>
        <ToggleButton value={tokenMap[CoinTypes.Bob]}> {CoinTypes.Bob} </ToggleButton>
      </ToggleButtonGroup>
      
}

export default ToggleComponent
