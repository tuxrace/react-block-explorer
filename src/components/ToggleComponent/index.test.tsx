import { render } from '@testing-library/react'
import ToggleComponent from './index'

describe('ToggleComponent component', () => {
    test('renders ToggleComponent component', async () => {
        const {baseElement} = render(<ToggleComponent />);
        expect(baseElement).toMatchSnapshot()
    })
})
