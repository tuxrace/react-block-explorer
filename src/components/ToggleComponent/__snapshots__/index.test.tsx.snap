// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar component renders Searchbar component 1`] = `
<body>
  <div>
    <div
      class="MuiToggleButtonGroup-root makeStyles-toggleStyle-1"
      role="group"
    >
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
        tabindex="0"
        type="button"
        value="5938fc8af82250ad6cf1da3bb92f4aa005cb2717"
      >
        <span
          class="MuiToggleButton-label"
        >
           
          AliceCoin
           
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
        tabindex="0"
        type="button"
        value="acfbdcf4693b1780e474116211f7f4795f24e06a"
      >
        <span
          class="MuiToggleButton-label"
        >
           
          BobCoin
           
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`ToggleComponent component renders ToggleComponent component 1`] = `
<body>
  <div>
    <div
      class="MuiToggleButtonGroup-root makeStyles-toggleStyle-1"
      role="group"
    >
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
        tabindex="0"
        type="button"
        value="5938fc8af82250ad6cf1da3bb92f4aa005cb2717"
      >
        <span
          class="MuiToggleButton-label"
        >
           
          AliceCoin
           
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
        tabindex="0"
        type="button"
        value="acfbdcf4693b1780e474116211f7f4795f24e06a"
      >
        <span
          class="MuiToggleButton-label"
        >
           
          BobCoin
           
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
